pkg/container/bitlpm/unsigned_test.go

```diff
@@ -0,0 +1,812 @@
+					}
+					rangeS := strings.Split(gotEntry, "-")
+					start, err := strconv.ParseUint(rangeS[0], 10, 16)
+					if err != nil {
+						t.Fatalf("Error parsing start value of range entry %q", gotEntry)
+					}
+					if uint16(start) > pr.start {
+						t.Fatalf("Expected to get an in range entry from key-prefix %d/%d, but got %q",
+							pr.start, pr.prefix(), gotEntry)
+					}
+					end, err := strconv.ParseUint(rangeS[1], 10, 16)
+					if err != nil {
+						t.Fatalf("Error parsing end value of range entry %q", gotEntry)
+					}
+					if uint16(end) < pr.end {
+						t.Fatalf("Expected to get an in range entry from key-prefix %d/%d, but got %q",
+							pr.start, pr.prefix(), gotEntry)
+					}
+				}
+			}
+		})
+	}
+}
+
+// TestUnsignedAncestors tests searching for all keys
+// that match a searched-for key and prefix.
+func TestUnsignedAncestors(t *testing.T) {
+	// Create a uint Trie that contains overlapping ranges
+	// from 0-65535.
+	tu := NewUintTrie[uint16, string]()
+	for i := uint(0); i < 16; i++ {
+		rng := uint16RangeMap[i]
+		entry := fmt.Sprintf("%d-%d", 0, rng)
+		tu.Upsert(i, rng, entry)
+	}
+	// Check to see that each range
+	// lookup returns all ranges that contain
+	// it.
+	for i := uint(0); i < 16; i++ {
+		rng := uint16RangeMap[i]
+		entry := fmt.Sprintf("%d-%d", 0, rng)
+		t.Run(entry, func(t *testing.T) {
+			expectedRes := make([]string, 0, i+1)
+			for t := uint(0); t <= i; t++ {
+				rng2 := uint16RangeMap[t]
+				expectedRes = append(expectedRes, fmt.Sprintf("%d-%d", 0, rng2))
+			}
+			gotRes := make([]string, 0, i+1)
+			tu.Ancestors(i, rng, func(prefix uint, key uint16, v string) bool {
+				gotRes = append(gotRes, v)
+				return true
+			})
+			if !reflect.DeepEqual(expectedRes, gotRes) {
+				t.Fatalf("Ancestors range %s, expected to get %v, but got: %v", entry, expectedRes, gotRes)
+			}
+		})
+	}
+}
+
+// TestUnsignedDescendants tests searching for all keys
+// that are match by a searched-for key and prefix.
+func TestUnsignedDescendants(t *testing.T) {
+	// Create a uint Trie that contains overlapping ranges
+	// from 0-65535.
+	tu := NewUintTrie[uint16, string]()
+	for i := uint(0); i <= 16; i++ {
+		rng := uint16RangeMap[i]
+		entry := fmt.Sprintf("%d-%d", 0, rng)
+		tu.Upsert(i, rng, entry)
+	}
+	// Check to see that each range lookup returns
+	// all ranges that contain it.
+	for i := uint(0); i < 16; i++ {
+		rng := uint16RangeMap[i]
+		entry := fmt.Sprintf("%d-%d", 0, rng)
+		t.Run(entry, func(t *testing.T) {
+			expectedRes := make([]string, 0, 16-i)
+			for t := i; t <= 16; t++ {
+				rng2 := uint16RangeMap[t]
+				expectedRes = append(expectedRes, fmt.Sprintf("%d-%d", 0, rng2))
+			}
+			gotRes := make([]string, 0, 16-i)
+			tu.Descendants(i, rng, func(prefix uint, key uint16, v string) bool {
+				gotRes = append(gotRes, v)
+				return true
+			})
+			if !reflect.DeepEqual(expectedRes, gotRes) {
+				t.Fatalf("Ancestors range %s, expected to get %v, but got: %v", entry, expectedRes, gotRes)
```

Comment:
This test refers to "ancestors" but it's testing Descendents. At minimum it seems to need a text update, but is there something else we need to double-check here?
```suggestion
				t.Fatalf("Descendents range %s, expected to get %v, but got: %v", entry, expectedRes, gotRes)
```